targeted/2xml4xxe.js

```diff
@@ -0,0 +1,248 @@
+	header.setHeader(org.parosproxy.paros.network.HttpRequestHeader.CONTENT_TYPE,"application/xml");
+	header.setHeader(org.parosproxy.paros.network.HttpRequestHeader.CONTENT_LENGTH,body.length);
+	msg.setRequestHeader(header);
+	ext = new org.parosproxy.paros.extension.history.ExtensionHistory;
+	man = ext.getResendDialog();
+	man.setMessage(msg.cloneRequest());
+	man.setVisible(true);
+}
+
+function isJson(str){
+	try{
+		JSON.parse(str);
+	}
+	catch(e){
+		return false;
+	}
+		return true;
+}
+
+function ismultipart(header){
+	type = header.getHeader(org.parosproxy.paros.network.HttpRequestHeader.CONTENT_TYPE);
+	if(type == null )
+		return false;
+	if(type.contains("multipart/form-data"))
+			return true;
+	return false;
+}
+
+function bodyToJson(body){
+	result = {};
+	tm = decodeURIComponent(body);
+	if(tm.indexOf('[') > -1 && tm.indexOf('[') > -1){
+		body = body.split('&');
+		for(i=0; i<body.length;i++){
+			body[i] = decodeURIComponent(body[i]);
+			out = "";
+			bdy = body[i];
+			opend = false;
+			len = bdy.length();
+			first = true;
+			for(j=0;j<len;j++){
+				if(first){
+					out += '{ "'
+					af = '' ;			
+					while(String.fromCharCode(bdy.charAt(j)) != '[' && String.fromCharCode(bdy.charAt(j)) != '=' ){
+						if(j == len)
+							break;
+						out += String.fromCharCode(bdy.charAt(j))
+						j++;
+					}
+					out += '" :';
+					af += '} ';
+					first = false;
+				}
+				if(String.fromCharCode(bdy.charAt(j)) == '=' && !opend){
+					j++;
+					out += '"'+bdy.substring(j,len)+'"';
+					break;
+					continue;
+				}	
+				else if(String.fromCharCode(bdy.charAt(j)) == '[' && !opend){
+					opend = true;
+					out += '{ "';
+					if(String.fromCharCode(bdy.charAt(j+1)) == ']' && String.fromCharCode(bdy.charAt(j+2)) == '='){
+						j += 2;
+						out += '"'+bdy.substring(j,len)+'"';
+						af += '} '
+						break;
+					}
+					continue;
+				}
+				else if(String.fromCharCode(bdy.charAt(j)) == ']'){
+					out += '" :';
+					af += " }"
+					opend = false;
+					continue;
+				}
+				else if(opend){
+					out += String.fromCharCode(bdy.charAt(j))
+					continue;
+				}
+				else {
+					//debug
+				}
+					
+			}
+				out += af
+				out = JSON.parse(out);
+			result = mergeJson(out,result);
+		}
+	}
+	else{
+		pairs = body.split('&');
+		pairs.forEach(function(i, pair){
+			pair = pair.split('=');
+			result[pair[0]] = decodeURIComponent(pair[1]||'');
+			});	
+	}
+return result;	
+}
+
+
+function jsonToXML(js){
+	xml = "";
+	for( key in js){
+		if( js[key] == null){
+			xml += js[key];
+		}
+		else if(typeof js[key] == "object"){
+			xml += toXml(key,jsonToXML(js[key]),1);
+		}
+		else {
+			xml += toXml(key,js[key],null);
+		}
+	}
+	return xml;
+}
+
+function toXml(key,value,att){ //pretify
+	if(att == null)
+		return ("<"+key+">"+value+"</"+key+">\n");
+	else{
+		if(value.slice(-1) == '\n')
+			value = value.substring(0,value.length-1);
+		return ("<"+key+">\n "+value+" \n</"+key+">\n");
+		}
+}
+function multiToJson(msg){
```

Comment:
For example, the [```if```/```else``` statement at line 178](https://github.com/zaproxy/community-scripts/pull/18/files#diff-d0be30a3da16ed672d3de040641a79e6R178), the ```else``` block was removed in the other script (```json_csrf_poc_generator.js```).